src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -1248,7 +1251,9 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 			let resolvedResult: { command: CommandString; args: CommandString[] } = await this.resolveCommandAndArgs(resolver, task.command);
 			command = resolvedResult.command;
 			args = resolvedResult.args;
-
+			if (options.cwd) {
+				task.updateWorkspaceFolder(options.cwd);
```

Comment:
Updating the workspace folder here is too late. Earlier, we will have already used the workspace folder for resolving variables, which means now those resolved variables might not match correctly.

https://github.dev/microsoft/vscode/blob/0f21601f5e5b95a0601d732dc2555776f263564d/src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts#L734-L735